### PR TITLE
Add secondary rate limit wait

### PR DESCRIPTION
**Pull request by EnricoMi**
Example log for primary rate limit error:

```
2022-12-16 13:56:47 +0000 - urllib3.connectionpool - DEBUG - https://api.github.com:443 "GET /repositories/290227969/commits/71ca27c4a9d74fa7ec2898304ee75909e4a57124/check-runs?per_page=1&page=150 HTTP/1.1" 403 None
2022-12-16 13:56:47 +0000 - github.GithubRetry -  INFO - Request GET /repositories/290227969/commits/71ca27c4a9d74fa7ec2898304ee75909e4a57124/check-runs?per_page=1&page=150 failed with 403: Forbidden
2022-12-16 13:56:47 +0000 - github.GithubRetry - DEBUG - There is no Retry-After in the response header
2022-12-16 13:56:47 +0000 - github.GithubRetry - DEBUG - Response body indicates retry-able rate limit error: API rate limit exceeded for installation ID 5425010.
2022-12-16 13:56:47 +0000 - urllib3.util.retry - DEBUG - Incremented Retry for (url='/repositories/290227969/commits/71ca27c4a9d74fa7ec2898304ee75909e4a57124/check-runs?per_page=1&page=150'): GithubRetry(total=9, connect=None, read=None, redirect=None, status=None)
2022-12-16 13:56:47 +0000 - github.GithubRetry - DEBUG - Reset occurs in 0:24:24.940607 (1671200472 / 2022-12-16 14:21:12)
2022-12-16 13:56:47 +0000 - github.GithubRetry - DEBUG - Setting next backoff to 1465.940607s
2022-12-16 14:21:13 +0000 - urllib3.connectionpool - DEBUG - Retry: /repositories/290227969/commits/71ca27c4a9d74fa7ec2898304ee75909e4a57124/check-runs?per_page=1&page=150
2022-12-16 14:21:13 +0000 - urllib3.connectionpool - DEBUG - Resetting dropped connection: api.github.com
2022-12-16 14:21:13 +0000 - urllib3.connectionpool - DEBUG - https://api.github.com:443 "GET /repositories/290227969/commits/71ca27c4a9d74fa7ec2898304ee75909e4a57124/check-runs?per_page=1&page=150 HTTP/1.1" 200 None
2022-12-16 14:21:13 +0000 - github.Requester - DEBUG - GET https://api.github.com/repositories/290227969/commits/71ca27c4a9d74fa7ec2898304ee75909e4a57124/check-runs?per_page=1&page=150 {'Accept': 'application/vnd.github.v3+json', 'Authorization': 'token (oauth token removed)', 'User-Agent': 'PyGithub/Python'} None ==> 200
```

Example log for secondary rate limit error, right before a primary rate limit reset:
```
2022-12-17 09:48:54 +0000 - urllib3.connectionpool - DEBUG - https://api.github.com:443 "GET /search/issues?q=type%3Apr+repo%3A%22EnricoMi%2Fpublish-unit-test-result-action%22+df23eea893bc760a02ff52d63d9ce4c6a06c381b&per_page=10 HTTP/1.1" 403 None
2022-12-17 09:48:54 +0000 - github.GithubRetry -  INFO - Request GET /search/issues?q=type%3Apr+repo%3A%22EnricoMi%2Fpublish-unit-test-result-action%22+df23eea893bc760a02ff52d63d9ce4c6a06c381b&per_page=10 failed with 403: Forbidden
2022-12-17 09:48:54 +0000 - github.GithubRetry - DEBUG - There is no Retry-After in the response header
2022-12-17 09:48:54 +0000 - github.GithubRetry - DEBUG - Response body indicates retry-able rate limit error: You have exceeded a secondary rate limit. Please wait a few minutes before you try again.
2022-12-17 09:48:54 +0000 - urllib3.util.retry - DEBUG - Incremented Retry for (url='/search/issues?q=type%3Apr+repo%3A%22EnricoMi%2Fpublish-unit-test-result-action%22+df23eea893bc760a02ff52d63d9ce4c6a06c381b&per_page=10'): GithubRetry(total=9, connect=None, read=None, redirect=None, status=None)
2022-12-17 09:48:54 +0000 - github.GithubRetry - DEBUG - Secondary rate limit has backoff of 60s
2022-12-17 09:48:54 +0000 - github.GithubRetry - DEBUG - Reset occurs in 0:00:00.476147 (1671270535 / 2022-12-17 09:48:55)
2022-12-17 09:48:54 +0000 - github.GithubRetry - DEBUG - Setting next backoff to 1.476147s
2022-12-17 09:48:56 +0000 - urllib3.connectionpool - DEBUG - Retry: /search/issues?q=type%3Apr+repo%3A%22EnricoMi%2Fpublish-unit-test-result-action%22+df23eea893bc760a02ff52d63d9ce4c6a06c381b&per_page=10
2022-12-17 09:48:56 +0000 - urllib3.connectionpool - DEBUG - https://api.github.com:443 "GET /search/issues?q=type%3Apr+repo%3A%22EnricoMi%2Fpublish-unit-test-result-action%22+df23eea893bc760a02ff52d63d9ce4c6a06c381b&per_page=10 HTTP/1.1" 403 None
2022-12-17 09:48:56 +0000 - github.GithubRetry -  INFO - Request GET /search/issues?q=type%3Apr+repo%3A%22EnricoMi%2Fpublish-unit-test-result-action%22+df23eea893bc760a02ff52d63d9ce4c6a06c381b&per_page=10 failed with 403: Forbidden
2022-12-17 09:48:56 +0000 - github.GithubRetry - DEBUG - There is no Retry-After in the response header
2022-12-17 09:48:56 +0000 - github.GithubRetry - DEBUG - Response body indicates retry-able rate limit error: You have exceeded a secondary rate limit. Please wait a few minutes before you try again.
2022-12-17 09:48:56 +0000 - urllib3.util.retry - DEBUG - Incremented Retry for (url='/search/issues?q=type%3Apr+repo%3A%22EnricoMi%2Fpublish-unit-test-result-action%22+df23eea893bc760a02ff52d63d9ce4c6a06c381b&per_page=10'): GithubRetry(total=8, connect=None, read=None, redirect=None, status=None)
2022-12-17 09:48:56 +0000 - github.GithubRetry - DEBUG - Secondary rate limit has backoff of 60s
2022-12-17 09:48:56 +0000 - github.GithubRetry - DEBUG - Reset occurs in 0:00:59.922572 (1671270596 / 2022-12-17 09:49:56)
2022-12-17 09:48:56 +0000 - github.GithubRetry - DEBUG - Setting next backoff to 60s
2022-12-17 09:49:56 +0000 - urllib3.connectionpool - DEBUG - Retry: /search/issues?q=type%3Apr+repo%3A%22EnricoMi%2Fpublish-unit-test-result-action%22+df23eea893bc760a02ff52d63d9ce4c6a06c381b&per_page=10
2022-12-17 09:49:56 +0000 - urllib3.connectionpool - DEBUG - https://api.github.com:443 "GET /search/issues?q=type%3Apr+repo%3A%22EnricoMi%2Fpublish-unit-test-result-action%22+df23eea893bc760a02ff52d63d9ce4c6a06c381b&per_page=10 HTTP/1.1" 200 None
```

Example log for primary rate limit error:

```
2022-12-16 13:56:47 +0000 - urllib3.connectionpool - DEBUG - https://api.github.com:443 "GET /repositories/290227969/commits/71ca27c4a9d74fa7ec2898304ee75909e4a57124/check-runs?per_page=1&page=150 HTTP/1.1" 403 None
2022-12-16 13:56:47 +0000 - github.GithubRetry -  INFO - Request GET /repositories/290227969/commits/71ca27c4a9d74fa7ec2898304ee75909e4a57124/check-runs?per_page=1&page=150 failed with 403: Forbidden
2022-12-16 13:56:47 +0000 - github.GithubRetry - DEBUG - Response body indicates retry-able primary rate limit error: API rate limit exceeded for installation ID 5425010.
2022-12-16 13:56:47 +0000 - urllib3.util.retry - DEBUG - Incremented Retry for (url='/repositories/290227969/commits/71ca27c4a9d74fa7ec2898304ee75909e4a57124/check-runs?per_page=1&page=150'): GithubRetry(total=9, connect=None, read=None, redirect=None, status=None)
2022-12-16 13:56:47 +0000 - github.GithubRetry - DEBUG - Reset occurs in 0:24:24.940607 (1671200472 / 2022-12-16 14:21:12)
2022-12-16 13:56:47 +0000 - github.GithubRetry - INFO - Setting next backoff to 1465.940607s
2022-12-16 14:21:13 +0000 - urllib3.connectionpool - DEBUG - Retry: /repositories/290227969/commits/71ca27c4a9d74fa7ec2898304ee75909e4a57124/check-runs?per_page=1&page=150
2022-12-16 14:21:13 +0000 - urllib3.connectionpool - DEBUG - Resetting dropped connection: api.github.com
2022-12-16 14:21:13 +0000 - urllib3.connectionpool - DEBUG - https://api.github.com:443 "GET /repositories/290227969/commits/71ca27c4a9d74fa7ec2898304ee75909e4a57124/check-runs?per_page=1&page=150 HTTP/1.1" 200 None
2022-12-16 14:21:13 +0000 - github.Requester - DEBUG - GET https://api.github.com/repositories/290227969/commits/71ca27c4a9d74fa7ec2898304ee75909e4a57124/check-runs?per_page=1&page=150 {'Accept': 'application/vnd.github.v3+json', 'Authorization': 'token (oauth token removed)', 'User-Agent': 'PyGithub/Python'} None ==> 200
```

Example log for secondary rate limit error, right before a primary rate limit reset:
```
2022-12-17 09:48:56 +0000 - urllib3.connectionpool - DEBUG - https://api.github.com:443 "GET /search/issues?q=type%3Apr+repo%3A%22EnricoMi%2Fpublish-unit-test-result-action%22+df23eea893bc760a02ff52d63d9ce4c6a06c381b&per_page=10 HTTP/1.1" 403 None
2022-12-17 09:48:56 +0000 - github.GithubRetry -  INFO - Request GET /search/issues?q=type%3Apr+repo%3A%22EnricoMi%2Fpublish-unit-test-result-action%22+df23eea893bc760a02ff52d63d9ce4c6a06c381b&per_page=10 failed with 403: Forbidden
2022-12-17 09:48:56 +0000 - github.GithubRetry - DEBUG - Response body indicates retry-able secondary rate limit error: You have exceeded a secondary rate limit. Please wait a few minutes before you try again.
2022-12-17 09:48:56 +0000 - urllib3.util.retry - DEBUG - Incremented Retry for (url='/search/issues?q=type%3Apr+repo%3A%22EnricoMi%2Fpublish-unit-test-result-action%22+df23eea893bc760a02ff52d63d9ce4c6a06c381b&per_page=10'): GithubRetry(total=8, connect=None, read=None, redirect=None, status=None)
2022-12-17 09:48:56 +0000 - github.GithubRetry - DEBUG - Setting next backoff to 60s
2022-12-17 09:49:56 +0000 - urllib3.connectionpool - DEBUG - Retry: /search/issues?q=type%3Apr+repo%3A%22EnricoMi%2Fpublish-unit-test-result-action%22+df23eea893bc760a02ff52d63d9ce4c6a06c381b&per_page=10
2022-12-17 09:49:56 +0000 - urllib3.connectionpool - DEBUG - https://api.github.com:443 "GET /search/issues?q=type%3Apr+repo%3A%22EnricoMi%2Fpublish-unit-test-result-action%22+df23eea893bc760a02ff52d63d9ce4c6a06c381b&per_page=10 HTTP/1.1" 200 None
```

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -9,7 +9,7 @@ LABEL com.github.actions.description="A GitHub Action to publish test results."
 LABEL com.github.actions.icon="check-circle"
 LABEL com.github.actions.color="green"
 
-RUN apk add --no-cache --upgrade expat libuuid
+RUN apk add --no-cache --upgrade expat libuuid git
```
[the remark this answers is not in the document]
Comment (EnricoMi) [reply Dockerfile:12]: @sonatype-lift ignore

**Comment by cmergenthaler** [comment]
Hi @EnricoMi, are there any plans on completing this? We are having secondary rate limit errors quite often.

**Comment by EnricoMi** [comment]
@cmergenthaler this code is already released and used. This PR is only about moving this into the PyGitHub project.

Please create a new issue with more details and I can take a look at it, if you are having rate limit errors with this action.

**Comment by cmergenthaler** [comment]
@EnricoMi alright: #419

**Comment by EnricoMi** [comment]
Moved to PyGithub 2.0.1rc0, which supports secondary rate limit.

**Pull request closed without merging**